ExpenseTracker/main.go

```diff
@@ -21,161 +23,188 @@ func main() {
 				fmt.Println(err)
+				continue
 			}
-			fmt.Printf("%-4s %-12s %-20s %s\n", "#", "Date", "Description", "Amount")
-			fmt.Println("--------------------------------------------------")
-			for _, v := range *expenses {
-				fmt.Printf("%-4v %-12v %-20v $%d\n",
-					v.Id(),
-					v.Date().Format("2006-01-02"),
-					v.Desc(),
-					v.Amount())
-			}
+			printList(expenses)
 
-		} else if cmd == "summary" {
+		case "summary":
 			summary, err := operator.Summary(conn)
 			if err != nil {
 				fmt.Println(err)
 				continue
-			} else {
-				fmt.Println(summary)
 			}
+			fmt.Println(summary)
 
-		} else if cmd == "help" {
+		case "help":
 			operator.Help()
 
-		} else if cmd == "delete" {
-			if len(fields) < 2 {
-				fmt.Println("You provided no id to delete")
-				operator.Help()
-				continue
-			}
-			identif, err := strconv.Atoi(fields[1])
+		case "delete":
+			err = deleteExpense(conn, fields)
 			if err != nil {
 				fmt.Println(err)
-			} else {
-				exists, err := operator.ExpenseExists(conn, identif)
-				if err != nil {
-					fmt.Println(err)
-					continue
-				}
-				if !exists {
-					fmt.Println("No expense with given id")
-					continue
-				}
-				err = operator.DeleteExpense(conn, identif)
-				if err != nil {
-					fmt.Println(err)
-					continue
-				} else {
-					fmt.Printf("Expense deleted succesfully (ID: %v)\n", identif)
-				}
-
-			}
-
-		} else if cmd == "update" {
-			if len(fields) < 4 {
-				fmt.Println("Not enough arguments provided for command \"update\"")
-				operator.Help()
 				continue
 			}
 
-			id, err := strconv.Atoi(fields[1])
+		case "update":
+			err = UpdateExpense(conn, fields)
 			if err != nil {
 				fmt.Println(err)
 				continue
 			}
 
-			exists, err := operator.ExpenseExists(conn, id)
+		case "reset":
+			err = operator.ResetDatabase(conn)
 			if err != nil {
 				fmt.Println(err)
 				continue
 			}
-			if !exists {
-				fmt.Println("No expense with provided id")
-				continue
-			}
+		default:
+			unknownCommand()
 
-			amount, err := strconv.Atoi(fields[3])
-			if err != nil {
-				fmt.Println(err)
-				continue
-			}
+		}
 
-			err = operator.ValidateData(fields[2], amount)
-			if err != nil {
-				fmt.Println(err)
-				continue
-			}
+	}
 
-			updatedExpense := operator.CreateExpense(fields[2], amount)
+}
 
-			err = operator.UpdateExpense(conn, id, updatedExpense)
-			if err != nil {
-				fmt.Println(err)
-			} else {
-				fmt.Printf("Succesfully updated ID: %v\n", id)
-			}
-		} else if cmd == "reset" {
-			err = operator.ResetDatabase(conn)
-			if err != nil {
-				fmt.Println(err)
-				continue
-			}
+func unknownCommand() {
+	fmt.Println("No such command!")
+	operator.Help()
+}
+
+func addExpense(conn *pgx.Conn, fields []string) error {
+	if len(fields) < 3 {
+		return fmt.Errorf("not enough arguments provided for command \"add\"")
+	}
+
+	temp, err := strconv.Atoi(fields[2])
+	if err != nil {
+		return err
+	}
+
+	err = operator.ValidateData(fields[1], temp)

```

Comment:
```suggestion
	description := fields[1]
	err = operator.ValidateData(description, temp)
```
Мне, как читателю, непонятно, что такое fields[1].  Ты, как автор, через месяц тоже не вспомнишь)